embedded/sql/row_reader.go

```diff
@@ -380,7 +381,11 @@ func (r *rawRowReader) reduceTxRange() (err error) {
 	return nil
 }
 
-func (r *rawRowReader) Read() (row *Row, err error) {
+func (r *rawRowReader) Read(ctx context.Context) (row *Row, err error) {
+	if ctx.Err() != nil {
```

Comment:
should this not be in a defer? Context could be cancelled during the whole read process

Reply:
Actually both checks are required, at the start and before returning

Reply:
discussed with you, closing this as the context is part of the interface hence included

Reply:
has discussed offline but leaving a comment for future reference, context might be closed at any point in time and it shouldn't make a difference in the result but it may avoid unnecessary delays whenever possible